File: ruby/hamming/hamming.rb
class Hamming
  def self.compute(s1, s2)
    raise ArgumentError if s1.size != s2.size
    (0..s1.size - 1).reduce(0) { s1[_2] != s2[_2] ? _1 + 1 : _1 }
  end
end

File: c/hamming/hamming.c
#include "hamming.h"
#include "string.h"

int compute(const char *lhs, const char *rhs) {


    if (strlen(lhs) != strlen(rhs)) return -1;
    int diff = 0;

    for (size_t i = 0; i < strlen(lhs); ++i) {
        if (lhs[i] != rhs[i]) ++diff;
    }

    return diff;
}
